core/src/main/clojure/xtdb/xtql.clj

```diff
@@ -116,6 +109,13 @@
              (set))
         expr/normalise-fn-name))
 
+(defn- plan-arg-bindings [args]
+  (let [arg-bindings (mapv plan-arg-spec args)
+        temporary-expr-symbols (map (comp #(*gensym* %) :l) arg-bindings)]
```

Comment:
I'd try to avoid using gensym directly, I think as a result you are missing a call to col-sym (and therefore metadata) here.

I'd use `gen-col` instead, feel free to extend to allow for a custom prefix string if you want the col to include the original col name.